File: backend/src/main/java/com/money/draft/service/AdminService.java
package com.money.draft.service;

import com.money.draft.domain.enums.AccountStatus;
import com.money.draft.dto.AdminAccountDetailResponse;
import com.money.draft.dto.AdminAccountView;
import com.money.draft.dto.AdminCreateAccountRequest;
import com.money.draft.dto.RewardLogResponse;
import com.money.draft.dto.TransactionLogResponse;
import java.util.List;

public interface AdminService {
    // Add these new signatures:
    AdminAccountDetailResponse createAccount(AdminCreateAccountRequest req);

    AdminAccountDetailResponse getAccountDetails(Long id);

    AdminAccountDetailResponse updateAccountStatus(Long id, AccountStatus status);

    // Existing signatures:
    List<AdminAccountView> getAllAccounts();
    List<TransactionLogResponse> getAllTransactions();
    List<RewardLogResponse> getAllRewards();
}

File: frontend/frontend/src/app/core/services/admin.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import {
  AdminAccountView,
  AdminAccountDetailResponse,
  AdminRewardDashboardResponse,
  TransactionDetailResponse,
  TransactionLogResponse,
  AdminCreateAccountRequest,
  AccountStatus,
  RewardTransactionResponse
} from '../models/api.models';

@Injectable({
  providedIn: 'root'
})
export class AdminService {
  private apiUrl = 'http://localhost:8080/admin';

  constructor(private http: HttpClient) {}

  getAllAccounts(): Observable<AdminAccountView[]> {
    return this.http.get<AdminAccountView[]>(`${this.apiUrl}/accounts`);
  }

  getAccountDetails(id: number): Observable<AdminAccountDetailResponse> {
    return this.http.get<AdminAccountDetailResponse>(`${this.apiUrl}/accounts/${id}`);
  }

  getAccountTransactions(id: number): Observable<TransactionLogResponse[]> {
    return this.http.get<TransactionLogResponse[]>(`${this.apiUrl}/transactions/${id}`);
  }

  createAccount(request: AdminCreateAccountRequest): Observable<AdminAccountDetailResponse> {
    return this.http.post<AdminAccountDetailResponse>(`${this.apiUrl}/accounts`, request);
  }

  updateAccountStatus(id: number, status: AccountStatus): Observable<AdminAccountDetailResponse> {
    const params = new HttpParams().set('status', status);
    return this.http.patch<AdminAccountDetailResponse>(`${this.apiUrl}/accounts/${id}/status`, null, { params });
  }

  getAllTransactions(): Observable<TransactionLogResponse[]> {
    return this.http.get<TransactionLogResponse[]>(`${this.apiUrl}/transactions`);
  }

  getRewardDashboard(): Observable<AdminRewardDashboardResponse> {
    return this.http.get<AdminRewardDashboardResponse>(`${this.apiUrl}/rewards/dashboard`);
  }

  getAllRewardTransactions(): Observable<RewardTransactionResponse[]> {
    return this.http.get<RewardTransactionResponse[]>(`${this.apiUrl}/rewards`);
  }

  getTransactionDetail(transactionId: number): Observable<TransactionDetailResponse> {
    return this.http.get<TransactionDetailResponse>(`${this.apiUrl}/transactions/detail/${transactionId}`);
  }
}
